didcomm/core/keys/authcrypt_keys_selector.py

```diff
@@ -38,11 +38,11 @@ async def find_authcrypt_pack_sender_and_recipient_keys(
         sender_did_doc = await resolvers_config.did_resolver.resolve(frm_did)
         if sender_did_doc is None:
             raise DIDDocNotResolvedError(frm_did)
-        if not sender_did_doc.key_agreement_kids:
+        if not sender_did_doc.key_agreement:
```

Comment:
How would this line (and others like it in this change set) be affected by an embedded verification method in the keyAgreement (or authentication) verification relationships? Do we expect that all DID Documents we expect to speak DIDComm with will always follow the pattern of verification methods defined in the `verificationMethod` attribute and only references will exist in the verification relationships?

Reply:
This might be a question @TelegramSam could help with

Reply:
Good question, but can we move this to an issue. This is already a big PR and it didn't change this behaviour.

Reply:
I'm not sure I totally agree; originally the DID resolver implementations must return a DID Doc instance that had the key_agreement_kids and authentication_kids, implying at least that the DID resolver should only return kids there. It's unclear what would have been done with embedded verification methods. But, using a DID Doc representation that more closely resembles the original resolved doc makes this a problem that needs to be more directly answered. But I'm fine to discuss further in an issue rather than in this PR.